crates/ntx-builder/src/builder/store.rs

```diff
@@ -0,0 +1,135 @@
+use std::net::SocketAddr;
+
+use miden_node_proto::{
+    errors::{ConversionError, MissingFieldHelper},
+    generated::{
+        requests::{
+            GetBlockHeaderByNumberRequest, GetNetworkAccountDetailsByPrefixRequest,
+            GetUnconsumedNetworkNotesRequest,
+        },
+        store::api_client as store_client,
+    },
+};
+use miden_node_utils::tracing::grpc::OtelInterceptor;
+use miden_objects::{
+    account::Account,
+    block::BlockHeader,
+    note::{Note, NoteTag},
+};
+use miden_tx::utils::Deserializable;
+use thiserror::Error;
+use tonic::{service::interceptor::InterceptedService, transport::Channel};
+use tracing::{info, instrument};
+
+use crate::COMPONENT;
+
+// STORE CLIENT
+// ================================================================================================
+
+type InnerClient = store_client::ApiClient<InterceptedService<Channel, OtelInterceptor>>;
+
+/// Interface to the store's gRPC API.
+///
+/// Essentially just a thin wrapper around the generated gRPC client which improves type safety.
+#[derive(Clone, Debug)]
+pub struct StoreClient {
+    inner: InnerClient,
+}
+
+impl StoreClient {
+    /// Creates a new store client with a lazy connection.
+    pub fn new(store_address: SocketAddr) -> Self {
```

Comment:
Should this take `Url` as a parameter instead of `SocketAddr`?

Reply:
Changed this to take `Url`